ERCS/erc-7744.md

```diff
@@ -0,0 +1,101 @@
+---
+eip: 7744
+title: Code Index
+description: Indexing smart contract bytecode on Ethereum.
+author: Tim Pechersky (@peersky) <t@peersky.xyz>
+discussions-to: https://ethereum-magicians.org/t/erc-7744-code-index/20569
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-07-16
+---
+
+## Abstract
+
+This EIP defines a standard interface for indexing smart contracts on Ethereum by their bytecode hash. This enables trustless discovery and verification of contract code, facilitating use cases like bytecode signing, whitelisting, and decentralized distribution mechanisms.
+
+## Motivation
+
+Existing contract discovery relies on addresses, which are non-deterministic and can be obfuscated through proxies. Indexing by bytecode hash provides a deterministic and tamper-proof way to identify and verify contract code, enhancing security and trust in the Ethereum ecosystem.
+
+Global bytecode identification is a requirement to build efficient at high scale global factories for future software distribution, that may want to rely on extensive re-use of same codebase as own dependencies.
+
```

Comment:
I'd like to see a bit more of an in-depth usecase here. Walk the reader through a concrete example of why someone would want to record code hashes on chain.

Reply:
added motivation